frontend/shared/src/components/attachments/AttachmentsList.tsx

```diff
@@ -1,60 +1,78 @@
-import { IconPlus } from 'hds-react';
+import { IconAlertCircleFill, IconPlus } from 'hds-react';
 import { useTranslation } from 'next-i18next';
 import * as React from 'react';
 import AttachmentItem from 'shared/components/attachments/AttachmentItem';
 import UploadAttachment from 'shared/components/attachments/UploadAttachment';
+import {
+  ATTACHMENT_CONTENT_TYPES,
+  ATTACHMENT_MAX_SIZE,
+} from 'shared/constants/attachment-constants';
 import Attachment from 'shared/types/attachment';
+import { getAttachmentsByType } from 'shared/utils/attachment.utils';
 
-import { $Container, $Heading, $Message } from './AttachmentsList.sc';
+import {
+  $Container,
+  $ErrorMessage,
+  $Heading,
+  $Message,
+} from './AttachmentsList.sc';
 
 type Props = {
   title: string;
   attachmentType: string;
-  allowedFileTypes: string[];
-  maxSize: number;
+  allowedFileTypes?: readonly string[];
+  maxSize?: number;
   message?: string | false;
+  errorMessage?: string | false;
   attachments?: Attachment[];
-  onUpload: (data: FormData) => void;
-  onRemove: (fileId: string) => void;
+  onUpload: (data: FormData) => void | Promise<void>;
+  onRemove: (fileId: string) => void | Promise<void>;
+  onOpen: (attachment: Attachment) => void | Promise<void>;
   isUploading: boolean;
   isRemoving: boolean;
+  required?: boolean;
+  buttonRef?: React.Ref<HTMLButtonElement>;
+  name?: string;
 };
 
 const AttachmentsList: React.FC<Props> = ({
   title,
   attachmentType,
-  allowedFileTypes,
-  maxSize,
+  allowedFileTypes = ATTACHMENT_CONTENT_TYPES,
+  maxSize = ATTACHMENT_MAX_SIZE,
   message,
+  errorMessage,
   attachments,
   onUpload,
   onRemove,
+  onOpen,
   isUploading,
   isRemoving,
+  required,
+  name,
+  buttonRef,
 }) => {
   const { t } = useTranslation();
   const translationsBase = 'common:applications.sections.attachments';
+
   const files = React.useMemo(
-    (): Attachment[] =>
-      attachments?.filter((att) => att.attachmentType === attachmentType) || [],
+    (): Attachment[] => getAttachmentsByType(attachments ?? [], attachmentType),
     [attachmentType, attachments]
   );
-
   return (
     <$Container>
-      <$Heading>{title}</$Heading>
+      <$Heading>
+        {title} {required && ' *'}
+      </$Heading>
```

Comment:
Nice, today this missing required asterisk was reported in our application to fix.